src/cryptoadvance/specter/controller.py

```diff
@@ -351,8 +351,14 @@ def general_settings():
                 as_attachment=True
             )
         elif action == "restore":
-            restore_devices = json.loads(request.form['restoredevices'])
-            restore_wallets = json.loads(request.form['restorewallets'])
+            if request.form['restoredevices']:
+                restore_devices = json.loads(request.form['restoredevices'])
+            else:
+                restore_devices = []
+            if request.form['restorewallets']:
+                restore_wallets = json.loads(request.form['restorewallets'])
+            else:
+                restore_wallets = []
```

Comment:
This is a bit shorter:

```suggestion
            restore_devices = json.loads(request.form.get("restoredevices", "[]"))
            restore_wallets = json.loads(request.form.get("restorewallets", "[]"))
```